### PR TITLE
Paradise Pop Effects

### PR DESCRIPTION
**Pull request by FalseIncarnate**
Adds effects to the recently added Paradise Pop reagents.
- This was not included in the initial PR since it affects balance and shouldn't hold up the base system which does not affect balance.

Effects by reagent:
- Paradise Punch: no effect, purely a flavor drink
- Apple-pocalypse: Very small chance to suck nearby objects towards the drinker each life tick (weak version of the liquid dark matter effect)
- Berry Banned:
 - Good version: Low chance to heal a tiny amount of a RANDOM damage type each cycle (less than omnizine, as well as less reliable)
 - Bad version: 50% chance to inflict toxin damage each cycle (same strength as the basic toxin reagent, but less frequent), BWOINKS the victim if they die with it in their system (more on this later)
 - image
- Blackeye Brew: Low chance to make the drinker shout a random bit of "tider talk" each cycle. Will make you sound like an idiot.
- Grape Granade: Low chance of making you burp, even lower chance of causing nearby objects to fly away from you (weaker version of sorium effect, same chance as the apple-pocalypse pull)
- Meteor Malt: Low chance to cause a screen shake and sound for only the drinker as if a meteor hit nearby, as well as a lower chance for a small amount (1-5 units) of a randomly selected mineral to appear in their bloodstream (mostly harmless, though there is a chance for it to be plasma)

CODER FUNTIME:
Adds support for death-triggered chem effects.
- Currently used only by the poisonous Berry Banned, this can be used in the future to have chems that could transform dead bodies into swarms of spiders, insta-borg a body, or even possibly revive a person the instant they die.

:cl:
rscadd: Adds effects to the Paradise Pop reagents. Drink them all to unlock their powers!
/:cl:

### CODE REVIEW

**Comment by FreeStylaLT** [comment]
This PR is getting fizzy!

**Comment by FreeStylaLT** [comment]
Let's see what False has brewed up this time!

**Comment by FreeStylaLT** [comment]
Let's have a toast to this PR!

**Comment by FreeStylaLT** [comment]
Ale make sure to upvote this!

**Comment by FreeStylaLT** [comment]
Let's give this one a pop!

**Comment by FreeStylaLT** [comment]
I don't want to see this PR               fizzle out!

**Comment by FreeStylaLT** [comment]
Look at what POPped up here!

**Comment by FreeStylaLT** [comment]
This PR has some flavor!

File: code/modules/reagents/newchem/paradise_pop.dm [hunk 1]
```diff
@@ -57,10 +124,36 @@
...
+/datum/reagent/grape_granade/on_mob_life(var/mob/living/M as mob)
+	if(!M)
+		M = holder.my_atom
+	if(prob(1))
+		var/turf/simulated/T = get_turf(M)
+		goonchem_vortex(T, 1, 1, 2)
+		M.emote("burp")
+		to_chat(M, "<span class='notice'>You feel ready to burst! Oh wait, just a burp...</span>")
+	else if(prob(25))
+		M.emote("burp")
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/paradise_pop.dm]: don't think humans have this as an emote, do they?

Comment (FreeStylaLT) [inline code/modules/reagents/newchem/paradise_pop.dm]: They do, last I checked

Comment (VampyrBytes) [inline code/modules/reagents/newchem/paradise_pop.dm]: Yep, they definitely do

Comment (FalseIncarnate) [inline code/modules/reagents/newchem/paradise_pop.dm]: They do and it works. "<name> burps." Is all it does, no sounds or anything, just a simple emote like gasp.

**Comment by EpicCharger** [comment]
@FreeStylaLT  After reading all these fancy drinks and the details, I was greeted with a ton of soda puns. Due to this I've basically forgotten what all the PR was about, and am only dying on the inside from too many puns.

**Comment by FreeStylaLT** [comment]
You could say I 7upped my game in sprite of it all.

**Comment by EpicCharger** [comment]
@FreeStylaLT  *Cries*

File: code/modules/reagents/newchem/newchem_procs.dm [hunk 2]
```diff
@@ -82,6 +82,19 @@
 				addiction_list.Remove(R)
 	update_total()
 
+/datum/reagents/proc/death_metabolize(var/mob/M)
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/newchem_procs.dm]: no `var/`

**Comment by Fox-McCloud** [comment]
Two things

no `var/`  for arguments and no `as mob`

File: code/modules/reagents/newchem/newchem_procs.dm [hunk 3]
```diff
@@ -82,6 +82,19 @@
 				addiction_list.Remove(R)
 	update_total()
 
+/datum/reagents/proc/death_metabolize(/mob/M)
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/newchem_procs.dm]: can't be `/mob/M` has to be `mob/M`. Same for all the others.

File: code/modules/reagents/newchem/newchem_procs.dm [hunk 4]
```diff
@@ -82,6 +82,19 @@
 				addiction_list.Remove(R)
 	update_total()
 
+/datum/reagents/proc/death_metabolize(mob/M)
+	if(M)
+		if(!istype(M, /mob/living))		//Non-living mobs can't metabolize reagents, so don't bother trying (runtime safety check)
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/newchem_procs.dm]: this is only ever called in carbon/death, so there's no point to this check

File: code/modules/reagents/oldchem/reagents/_reagent_base.dm [hunk 5]
```diff
@@ -69,6 +69,9 @@
 		holder.remove_reagent(id, metabolization_rate) //By default it slowly disappears.
 		current_cycle++
 
+/datum/reagent/proc/on_mob_death(var/mob/living/M as mob, var/alien)	//use this to have chems have a "death-triggered" effect
```
Comment (Fox-McCloud) [inline code/modules/reagents/oldchem/reagents/_reagent_base.dm]: still need to get rid of teh var here and as mob.

Also the alien argument is completely dead now.

File: code/modules/reagents/newchem/paradise_pop.dm [hunk 6]
```diff
@@ -33,6 +42,29 @@
 	reagent_state = LIQUID
 	color = "#FF44FF"
 
+/datum/reagent/berry_banned/on_mob_life(mob/living/M)
+	if(!M)
+		M = holder.my_atom
+	if(!iscarbon(M))
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/paradise_pop.dm]: why the carbon check?

File: code/modules/reagents/newchem/paradise_pop.dm [hunk 7]
```diff
@@ -49,6 +97,25 @@
 	reagent_state = LIQUID
 	color = "#4d2600"
 
+/datum/reagent/blackeye_brew/on_mob_life(mob/living/M)
+	if(!M)
+		M = holder.my_atom
+	if(!ishuman(M))
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/paradise_pop.dm]: Come on, I want my greytider Xenos :wink:

File: code/modules/reagents/newchem/paradise_pop.dm [hunk 8]
```diff
@@ -57,10 +124,36 @@
...
 	name = "Meteor Malt"
 	id = "meteor_malt"
 	description = "Soft drinks have been detected on collision course with your tastebuds."
 	reagent_state = LIQUID
 	color = "#cc9900"
+
+/datum/reagent/meteor_malt/on_mob_life(mob/living/M)
+	if(!M)
+		M = holder.my_atom
+	if(!iscarbon(M))
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/paradise_pop.dm]: again, why the carbon checks?

File: code/modules/reagents/newchem/newchem_procs.dm [hunk 9]
```diff
@@ -82,6 +82,19 @@
...
+			return
+		if(M.stat != DEAD)				//what part of DEATH_metabolize don't you get?
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/newchem_procs.dm]: make sure this works for carbons---pretty sure it'll work with humans.

File: code/modules/reagents/newchem/paradise_pop.dm [hunk 10]
```diff
@@ -33,6 +42,29 @@
 	reagent_state = LIQUID
 	color = "#FF44FF"
 
+/datum/reagent/berry_banned/on_mob_life(mob/living/M)
+	if(!M)
+		M = holder.my_atom
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/paradise_pop.dm]: all of these
```
	if(!M)
	     M = holder.my_atom
```

Can be removed

File: code/modules/reagents/newchem/newchem_procs.dm [hunk 11]
```diff
@@ -82,6 +82,19 @@
...
+			return
+	for(var/A in reagent_list)
+		var/datum/reagent/R = A
+		if(!istype(R))
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/newchem_procs.dm]: I'd really like to know how non-reagents end up in the reagent list.

File: code/modules/reagents/newchem/newchem_procs.dm (hunk 9 above)
Comment (FalseIncarnate) [inline code/modules/reagents/newchem/newchem_procs.dm]: tested with a xeno, does in fact work

File: code/modules/reagents/newchem/newchem_procs.dm [hunk 12]
```diff
@@ -12,12 +12,11 @@
 	var/chem_temp = 300
 	var/list/datum/reagent/addiction_list = new/list()
 
-/datum/reagents/proc/metabolize(var/mob/M)
-	if(M)
-		if(!istype(M, /mob/living))		//Non-living mobs can't metabolize reagents, so don't bother trying (runtime safety check)
-			return
-		chem_temp = M.bodytemperature
-		handle_reactions()
+/datum/reagents/proc/metabolize(mob/M)
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/newchem_procs.dm]: leave this be; this is addressed in my reagents cleanup PR.

File: code/modules/reagents/newchem/newchem_procs.dm [hunk 13]
```diff
@@ -82,6 +82,18 @@
 				addiction_list.Remove(R)
 	update_total()
 
+/datum/reagents/proc/death_metabolize(mob/M)
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/newchem_procs.dm]: typecast this for `mob/living/M`

File: code/modules/reagents/newchem/paradise_pop.dm [hunk 14]
```diff
@@ -33,6 +40,25 @@
 	reagent_state = LIQUID
 	color = "#FF44FF"
 
+/datum/reagent/berry_banned/on_mob_life(mob/living/M)
+	if(prob(10))
+		var/heal_type = pick("brute", "fire", "toxin", "oxygen", "clone", "brain")
```
Comment (Fox-McCloud) [inline code/modules/reagents/newchem/paradise_pop.dm]: numbers over stringssss

**Pull request merged**